docs/ref/models/constraints.txt

```
@@ -282,27 +282,43 @@ PostgreSQL 15+.
 
 .. attribute:: UniqueConstraint.violation_error_code
 
-The error code used when ``ValidationError`` is raised during
-:ref:`model validation <validating-objects>`. Defaults to ``None``.
+The error code used when a ``ValidationError`` is raised during
+:ref:`model validation <validating-objects>`.
+
+If :attr:`~UniqueConstraint.fields` is set and without a
+:attr:`~UniqueConstraint.condition`, the
+:attr:`~django.db.models.UniqueConstraint.violation_error_code` defaults to the
+same error code as when setting :attr:`.Field.unique` when there is a single
+field, and to the same error code as when setting
+:attr:`Meta.unique_together <django.db.models.Options.unique_together>` when
+there are multiple.
 
-This code is *not used* for :class:`UniqueConstraint`\s with
-:attr:`~UniqueConstraint.fields` and without a
-:attr:`~UniqueConstraint.condition`. Such :class:`~UniqueConstraint`\s have the
-same error code as constraints defined with :attr:`.Field.unique` or in
-:attr:`Meta.unique_together <django.db.models.Options.constraints>`.
+Otherwise, defaults to :attr:`.BaseConstraint.violation_error_message`.
+
+.. versionchanged:: 5.2
+
+If :attr:`~django.db.models.UniqueConstraint.violation_error_code` is
+set, this code is now always used.
 
 ``violation_error_message``
 ---------------------------
 
 .. attribute:: UniqueConstraint.violation_error_message
 
-The error message used when ``ValidationError`` is raised during
-:ref:`model validation <validating-objects>`. Defaults to
-:attr:`.BaseConstraint.violation_error_message`.
-
-This message is *not used* for :class:`UniqueConstraint`\s with
-:attr:`~UniqueConstraint.fields` and without a
-:attr:`~UniqueConstraint.condition`. Such :class:`~UniqueConstraint`\s show the
-same message as constraints defined with
-:attr:`.Field.unique` or in
-:attr:`Meta.unique_together <django.db.models.Options.constraints>`.
+The error message used when a ``ValidationError`` is raised during
+:ref:`model validation <validating-objects>`.
+
+If :attr:`~UniqueConstraint.fields` is set and without a
+:attr:`~UniqueConstraint.condition`, the
+:attr:`~django.db.models.UniqueConstraint.violation_error_message` defaults to
+the same error message as when setting :attr:`.Field.unique` when there is a
+single field, and to the same error message as when setting
+:attr:`Meta.unique_together <django.db.models.Options.unique_together>` when
+there are multiple.
+
+Otherwise, defaults to :attr:`.BaseConstraint.violation_error_message`.
```

This is still quite a mouth-full. Do you think this is clearer?
```suggestion
Defaults to :attr:`.BaseConstraint.violation_error_message`, when either
:attr:`.UniqueConstraint.condition` is provided or
:attr:`.UniqueConstraint.fields` is not provided.

When :attr:`.UniqueConstraint.fields` is provided without a
:attr:`.UniqueConstraint.condition`, then if :attr:`.UniqueConstraint.fields`
references multiple fields, defaults to the
:attr:`Meta.unique_together <django.db.models.Options.unique_together>` error
message. Otherwise defaults to the :attr:`.Field.unique` error message.
```